Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -1682,9 +1707,14 @@ void CachedResourceLoader::warnUnusedPreloads()
 {
     if (!m_preloads)
         return;
+
+    auto* document = this->document();
+    if (!document)
+        return;
+
     for (const auto& resource : *m_preloads) {
-        if (resource.isLinkPreload() && resource.preloadResult() == CachedResource::PreloadResult::PreloadNotReferenced && document()) {
-            document()->addConsoleMessage(MessageSource::Other, MessageLevel::Warning,
+        if (resource.isLinkPreload() && resource.preloadResult() == CachedResource::PreloadResult::PreloadNotReferenced && document) {
```

Comment:
... so the null check here for every resource does nothing.

Maybe you intended to store the document in a WeakPtr?

Reply:
No -- I missed these cases. I can remove them. document() is already a week pointer, and there should be no script running in these calls to change that state.